File: VideoPlayer/VideoPlayer/ViewController.m
//
//  ViewController.m
//  VideoPlayer
//
//  Created by delphiwu on 16/9/2.
//  Copyright © 2016年 delphi. All rights reserved.
//

#import "ViewController.h"
#import "DFPlayer.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet DFPlayer *dfPlayer;
@property (weak, nonatomic) IBOutlet UIImageView *cover;
@property (weak, nonatomic) IBOutlet UIButton *playBtn;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    [[NSNotificationCenter defaultCenter]addObserver:self selector:@selector(playerDidEndPlay) name:DFPlayerFinishedPlayNotificationKey object:nil];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)playBtnPressed {
    
    _cover.hidden = YES;
    _playBtn.hidden = YES;
    
    _dfPlayer.status = ^(DFPlayerStatusChangeType type){
        /*
         DFPlayerStatusLoadingType           = 0, //正在加载
         DFPlayerStatusReadyToPlayType       = 1, //开始播放
         DFPlayerStatusrLoadedTimeRangesType = 2  //开始缓存
         */
    };
    [_dfPlayer playWithVideoUrl:@"http://flv2.bn.netease.com/videolib3/1609/02/YJVxD6726/SD/YJVxD6726-mobile.mp4"];
    
}

-(void)playerDidEndPlay {
    _cover.hidden = NO;
    _playBtn.hidden = NO;
}

-(void)dealloc {
    [[NSNotificationCenter defaultCenter]removeObserver:self];
}

@end

File: PingTransitionDemo/PingTransitionDemo/ViewController.swift
//
//  ViewController.swift
//  PingTransitionDemo
//
//  Created by delphiwu on 16/9/2.
//  Copyright © 2016年 delphi. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    let animation = PingTransitionAnimation()
    var touchPoint = CGPointZero
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        navigationController?.delegate = self
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func presentBtnPressed(sender: AnyObject) {
        let vc = UIStoryboard.init(name: "Main", bundle: nil).instantiateViewControllerWithIdentifier("PresentedViewController") as! PresentedViewController
        
        vc.transitioningDelegate = self
        vc.modalPresentationStyle = .FullScreen
        presentViewController(vc, animated: true, completion: nil)
    }
    
}

//MARK:UINavigationControllerDelegate
extension ViewController:UINavigationControllerDelegate {
    func navigationController(navigationController: UINavigationController, animationControllerForOperation operation: UINavigationControllerOperation, fromViewController fromVC: UIViewController, toViewController toVC: UIViewController) -> UIViewControllerAnimatedTransitioning? {
        if operation == .Push {
            animation.transitionMode = .open
            animation.duration = 0.5
            animation.startPoint = touchPoint
            return animation
        } else if operation == .Pop {
            animation.transitionMode = .close
            animation.duration = 0.5
            animation.startPoint = touchPoint
            return animation
        }else {
            return nil
        }
    }
}

//MARK:UIViewControllerTransitioningDelegate
extension ViewController:UIViewControllerTransitioningDelegate {
    func animationControllerForPresentedController(presented: UIViewController, presentingController presenting: UIViewController, sourceController source: UIViewController) -> UIViewControllerAnimatedTransitioning? {
        animation.duration = 0.5
        animation.transitionMode = .open
        animation.startPoint = touchPoint
        return animation
    }
    
    func animationControllerForDismissedController(dismissed: UIViewController) -> UIViewControllerAnimatedTransitioning? {
        animation.duration = 0.5
        animation.transitionMode = .close
        animation.startPoint = touchPoint
        return animation
    }
}

//MARK:UIGestureRecognizerDelegate
extension ViewController:UIGestureRecognizerDelegate {
    func gestureRecognizer(gestureRecognizer: UIGestureRecognizer, shouldReceiveTouch touch: UITouch) -> Bool {
        touchPoint = touch.locationInView(self.view)
        return false
    }
}

